### PR TITLE
Update Pull Request template: handling CHANGELOG.md

### PR DESCRIPTION
**Pull request by joergsteffens**
#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General
- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
- [x] Commit descriptions are understandable and well formatted

##### Source code quality

- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems
- [x] `git status` should not report modifications in the source tree after building and testing

### CODE REVIEW

File: docs/manuals/source/DeveloperGuide/ReleasingBareos.rst [hunk 1]
```diff
@@ -34,8 +34,8 @@ There are version-dependent files in the Bareos sources that might need your att
 
 version.map.in
 ~~~~~~~~~~~~~~
-The file is in either :file:`core/src/cats/ddl` or :file:`src/cats/ddl`.
-Whenever the database schema version changes this file must be updated.
+The file :file:`core/src/cats/ddl/version.map.in` must be updated,
```
Comment (pstorz) [inline docs/manuals/source/DeveloperGuide/ReleasingBareos.rst:37]:
```suggestion
The file :file:`core/src/cats/ddl/version.map.in` must be updated
```

File: docs/manuals/source/DeveloperGuide/generaldevel.rst [hunk 2]
```diff
@@ -130,6 +130,17 @@ The configuration file will be automatically found and used by clang-format:
   #format one sourcecode file in-place
   clang-format -i ./core/src/dird/dird_conf.cc
 
+The Bareos has bundled its automatic sourcecode formatting into one tool: ``bareos-check-sources``.
```
Comment (pstorz) [inline docs/manuals/source/DeveloperGuide/generaldevel.rst:133]:
```suggestion
The Bareos project has bundled its automatic sourcecode formatting into one tool: ``bareos-check-sources``.
```

File: pull_request_template.md [hunk 3]
```diff
@@ -19,9 +19,8 @@ If you have any questions or problems, please give a comment in the PR.
 ##### General
 - [ ] PR name is meaningful
 - [ ] Purpose of the PR is understood
-- [ ] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
```
Comment (pstorz) [inline pull_request_template.md:22]: I would like to keep one line for the CHANGELOG.md checks and not split it up into 4 single points.

File: pull_request_template.md [hunk 4]
```diff
@@ -38,3 +37,9 @@ If you have any questions or problems, please give a comment in the PR.
 - [ ] The choice of the type of test (unit test or systemtest) is reasonable
 - [ ] Testname matches exactly what is being tested
 - [ ] On a fail, output of the test leads quickly to the origin of the fault
+
+##### CHANGELOG.md
+- [ ] Separate commit for this PR in the CHANGELOG.md
+- [ ] The changelog entries refers to this PR (verify the number)
+- [ ] New changelog entries are added at the top of a section.
+- [ ] Commit message is "updated CHANGELOG.md" or "updated CHANGELOG.md for PR #0000" (replace 0000 by the number of the current PR)
```
Comment (pstorz) [inline pull_request_template.md:45]:
```suggestion
- [ ] Commit message is "update CHANGELOG.md" or "update CHANGELOG.md for PR #0000" (replace 0000 by the number of the current PR)
```

Comment (pstorz) [reply pull_request_template.md:45]: refer to the developer guideline for git commits: use imperative language.

But as said above, I would not want to split the CHANGELOG.md checkmark into four submarks.

**Review by pstorz — CHANGES_REQUESTED** [review]

File: pull_request_template.md (hunk 3 above)
Comment (joergsteffens) [reply pull_request_template.md:22]: I changed it back to a single line and added a section in the Developer Guide.
I also changed "PR number referenced is same" into "PR number is verified.". I considered the initial wording misleading, however I'm also not to happy with my replacement. Maybe you got a better wording?

Comment (bruno-at-bareos) [reply pull_request_template.md:22]: I propose
```diff
+- [ ] A separate commit in CHANGELOG.md with this PR number exist and is placed on top of the section
```

File: pull_request_template.md [hunk 5]
```diff
@@ -19,9 +19,9 @@ If you have any questions or problems, please give a comment in the PR.
 ##### General
 - [ ] PR name is meaningful
 - [ ] Purpose of the PR is understood
-- [ ] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
 - [ ] Commit descriptions are understandable and well formatted
-- [ ] If backport: add original PR number and target branch at top of this file: **Backport of PR#000 to bareos-2x**
+- [ ] Separate commit for CHANGELOG.md changes (message: "update CHANGELOG.md" or "update CHANGELOG.md for PR #0000"). PR number is verified.
```
Comment (bruno-at-bareos) [inline pull_request_template.md:23]:
```suggestion
+- [ ] Separate commit exists for CHANGELOG.md with this PR changes and number placed on top of corresponding section. (message: "update CHANGELOG.md" or "update CHANGELOG.md for PR #0000")
```

File: docs/manuals/source/DeveloperGuide/gitworkflow.rst [hunk 6]
```diff
@@ -79,3 +87,22 @@ It should be wrapped at 72 chars and can consist of multiple paragraphs separate
...
+CHANGELOG
+---------
+
+Each pull-request (PR) should have an entry in the CHANGELOG.md file.
+
+The entry in the CHANGELOG.md file should
+
+* be in a separate commit.
+* refers to the PR.
+* be added at the top of the relevant section.
```
Comment (pstorz) [inline docs/manuals/source/DeveloperGuide/gitworkflow.rst:101]:
```suggestion
* be added at the bottom of the relevant section.
```

File: docs/manuals/source/DeveloperGuide/gitworkflow.rst [hunk 7]
```diff
@@ -79,3 +87,22 @@ It should be wrapped at 72 chars and can consist of multiple paragraphs separate
...
+
+The commit message should be:
+
+* ``update CHANGELOG.md`` or
+* ``update CHANGELOG.md for PR #0000``
```
Comment (pstorz) [inline docs/manuals/source/DeveloperGuide/gitworkflow.rst:106]:
```suggestion
```

File: pull_request_template.md (hunk 5 above)
Comment (pstorz) [inline pull_request_template.md:23]:
```suggestion
- [ ] Separate commit for CHANGELOG.md ("update CHANGELOG.md"). PR number is correct.
```

File: pull_request_template.md [hunk 8]
```diff
@@ -19,9 +19,9 @@ If you have any questions or problems, please give a comment in the PR.
...
+- [ ] If backport: add original PR number and target branch at top of this file: **Backport of PR #0000 to bareos-2x**
```
Comment (pstorz) [inline pull_request_template.md:24]:
```suggestion
- [ ] check backport line
``` 
Add this to top of file: **Backport of PR #0000 to bareos-2x** (remove if no backport)

File: pull_request_template.md [hunk 9]
```diff
@@ -30,7 +32,6 @@ If you have any questions or problems, please give a comment in the PR.
 - [ ] Code comments are correct (logically and spelling)
 - [ ] Required documentation changes are present and part of the PR
 - [ ] `bareos-check-sources --since-merge` does not report any problems
-- [ ] `git status` should not report modifications in the source tree after building and testing
```
Comment (bruno-at-bareos) [inline pull_request_template.md:33]: Why did we remove that line?

Comment (joergsteffens) [reply pull_request_template.md:33]: @pstorz said, that this have not happened for a long time and can therefore be removed.

Comment (bruno-at-bareos) [reply pull_request_template.md:33]: I would have keep it, especially for external contribution, but we can try without and see.

**Pull request merged**